`var x: seq[array[1, int]]` causes 'conflicting types' compile error from GCC

mytests/test.nim:
```nim
var x: seq[array[1, int]]
```

```console
$ ./bin/nimony c -r mytests/test.nim 
bin/nimsem  --isSystem m nimcache/sysvq0asl.1.nif nimcache/sysvq0asl.2.nif nimcache/sysvq0asl.2.idx.nif
bin/nimsem  m nimcache/tes9hvujc.1.nif nimcache/tes9hvujc.2.nif nimcache/tes9hvujc.2.idx.nif
cc -c   -Ivendor/mimalloc/include vendor/mimalloc/src/static.c -o nimcache/static.o
bin/hexer --bits:64 nimcache/tes9hvujc.2.nif
bin/nifc c --compileOnly --isMain  nimcache/tes9hvujc.c.nif
cc -c  -Imytests  nimcache/tes9hvujc.c -o nimcache/tes9hvujc.o
nimcache/tes9hvujc.c:325:1: error: conflicting types for ‘AarrayAiSminusQ1ZS1_0_t’; have ‘struct <anonymous>’
  325 | AarrayAiSminusQ1ZS1_0_t;
      | ^~~~~~~~~~~~~~~~~~~~~~~
nimcache/tes9hvujc.c:318:40: note: previous declaration of ‘AarrayAiSminusQ1ZS1_0_t’ with type ‘AarrayAiSminusQ1ZS1_0_t’
  318 | typedef struct AarrayAiSminusQ1ZS1_0_t AarrayAiSminusQ1ZS1_0_t;
      |                                        ^~~~~~~~~~~~~~~~~~~~~~~
```

In generated C code, there are two typedef for `AarrayAiSminusQ1ZS1_0_t`.
```c
typedef struct AarrayAiSminusQ1ZS1_0_t AarrayAiSminusQ1ZS1_0_t;
typedef struct seq_0_tes9hvujc{
  NI len_0_tes9hvujc;
  AarrayAiSminusQ1ZS1_0_t* data_0_tes9hvujc;}
seq_0_tes9hvujc;
typedef struct {
  NI a[1];}
AarrayAiSminusQ1ZS1_0_t;
```

I tested https://github.com/nim-lang/nimony/pull/707.
I still get compile error from gcc:
```console
nimcache/tes9hvujc.c:325:1: error: conflicting types for ‘AarrayAiSminusQ1ZS1_0_t’; have ‘struct <anonymous>’
  325 | AarrayAiSminusQ1ZS1_0_t;
      | ^~~~~~~~~~~~~~~~~~~~~~~
nimcache/tes9hvujc.c:318:43: note: previous declaration of ‘AarrayAiSminusQ1ZS1_0_t’ with type ‘AarrayAiSminusQ1ZS1_0_t’ {aka ‘struct Q1_Aarr
ayAiSminusQ1ZS1_0_t’}
  318 | typedef struct Q1_AarrayAiSminusQ1ZS1_0_t AarrayAiSminusQ1ZS1_0_t;
      |                                           ^~~~~~~~~~~~~~~~~~~~~~~
```

--- a/src/nifc/gentypes.nim
+++ b/src/nifc/gentypes.nim
@@ -456,7 +456,6 @@ proc generateTypes(c: var GeneratedCode; o: TypeOrder) =
     let decl = takeTypeDecl(n)
     let s = mangle(pool.syms[decl.name.symId])
     c.add declKeyword
-    c.add "Q1_"
     c.add s
     c.add Space
     c.add s
@@ -470,6 +469,7 @@ proc generateTypes(c: var GeneratedCode; o: TypeOrder) =
       case decl.body.typeKind
       of ArrayT:
         c.add declKeyword
+        c.add s
         c.add CurlyLe
         var n = decl.body.firstSon
         genType c, n, "a"
@@ -491,7 +491,6 @@ proc generateTypes(c: var GeneratedCode; o: TypeOrder) =
         c.add Semicolon
       else:
         c.add declKeyword
-        c.add "Q1_"
         c.add s
         c.add CurlyLe
         # XXX generate attributes and pragmas here

